src/view/wlr-surface-node.cpp

```diff
@@ -508,5 +510,16 @@ void wf::scene::wlr_surface_node_t::update_pending_outputs()
         }
     }
 
+    if (entered)
```

Comment:
We need to update not only when a new output is entered, but also when we send a leave event. It could be that the output with highest scale was removed and the rest remain visible (this could easily happen if you use the move plugin to drag a view from output with high to low scale factor).